src/AddTodoForm.js

```diff
@@ -1,18 +1,48 @@
 import React from "react";
 
 export default function AddTodoForm(props) {
+
+  // const {onAddTodo, todoTitle} = props; // applying destructuring
+
+  const [todoTitle, setTodoTitle] = React.useState("");
+
+  const handleTitleChange = (event) => {
+    let newTodoTitle = event.target.value;
+    setTodoTitle(newTodoTitle);
+    // console.log("newTodoTitle = " + newTodoTitle);
+  };
+
+  /// ----------------------------- ///
+
   const handleAddTodo = (event) => {
+
     event.preventDefault();
-    let todoTitle = event.target.title.value;
-    // console.log("todoTitle = " + todoTitle);
-    props.onAddTodo(todoTitle);
+    // let todoTitle = event.target.title.value; // old
+    // props.onAddTodo(todoTitle); // old
+
+    props.onAddTodo({ title: todoTitle, id: Date.now() });
+   
+   // Inside handleAddTodo, remove the reset() method and replace it with logic to reset the todoTitle state to an empty String
+
+    event.target[0].value="" // is this correct logic? If some mentor could verify in his feedback. Thx
```

Comment:
This technically works because when you enter in another todo item into input it runs through the handleTitleChange function and calls setTodoTitle. I think the correct logic was was to call setTodoTitle(""); with an empty string.